cmd/package_managers.go

```diff
@@ -138,3 +205,145 @@ func fetchGitRepositoryFromRubyGems(packageName string, manager packageManagerCl
 	}
 	return v.SourceCodeURI, nil
 }
+
+// Gets the GitHub repository URL for the nuget package.
+func fetchGitRepositoryFromNuget(packageName string, manager packageManagerClient) (string, error) {
+	nugetIndexURL := "https://api.nuget.org/v3/index.json"
+	respIndex, err := manager.GetURI(nugetIndexURL)
+	if err != nil {
+		return "", sce.WithMessage(sce.ErrScorecardInternal, fmt.Sprintf("failed to get nuget index json: %v", err))
+	}
+	defer respIndex.Body.Close()
+	nugetIndexResults := &nugetIndexResults{}
+	err = json.NewDecoder(respIndex.Body).Decode(nugetIndexResults)
+
+	if err != nil {
+		return "", sce.WithMessage(sce.ErrScorecardInternal,
+			fmt.Sprintf("failed to parse nuget index json: %v", err))
+	}
+	nugetPackageBaseURL, err := getFieldFromIndexResults(nugetIndexResults.Resources,
+		"PackageBaseAddress/3.0.0")
+	if err != nil {
+		return "", err
+	}
+	nugetRegistrationBaseURL, err := getFieldFromIndexResults(nugetIndexResults.Resources,
+		"RegistrationsBaseUrl/3.6.0")
+	if err != nil {
+		return "", err
+	}
+
+	lowerCasePackageName := strings.ToLower(packageName)
+	lastPackageVersion, err := getLatestListedVersion(nugetRegistrationBaseURL,
+		lowerCasePackageName, manager)
+	if err != nil {
+		return "", err
+	}
+
+	respPackageSpec, err := manager.Get(
+		nugetPackageBaseURL+"%[1]v/"+lastPackageVersion+"/%[1]v.nuspec", lowerCasePackageName)
+	if err != nil {
+		return "", sce.WithMessage(sce.ErrScorecardInternal,
+			fmt.Sprintf("failed to get nuget package spec json: %v", err))
+	}
+	defer respPackageSpec.Body.Close()
+
+	packageSpecResults := &nugetNuspec{}
+	err = xml.NewDecoder(respPackageSpec.Body).Decode(packageSpecResults)
+
+	if err != nil || packageSpecResults.Metadata == (nuspecMetadata{}) {
```

Comment:
optional, no parens (and other occurrences)

Reply:
also, err can be nil here, which can make the error message below confusing if the problem is really an empty response. it may be worth breaking into 2 checks

Reply:
fixed the second comment. not sure what you mean by no parens in this case. since this doesn't compile:
packageSpecResults.Metadata == nuspecMetadata{} {
...
}

Reply:
Yes, you're right. ignore that.